homeassistant/components/knx/switch.py

```diff
@@ -82,3 +85,52 @@ async def async_turn_on(self, **kwargs: Any) -> None:
     async def async_turn_off(self, **kwargs: Any) -> None:
         """Turn the device off."""
         await self._device.set_off()
+
+
+class KnxYamlSwitch(_KnxSwitch):
+    """Representation of a KNX switch configured from YAML."""
+
+    def __init__(self, xknx: XKNX, config: ConfigType) -> None:
+        """Initialize of KNX switch."""
+        super().__init__(
+            device=XknxSwitch(
+                xknx,
+                name=config[CONF_NAME],
+                group_address=config[KNX_ADDRESS],
+                group_address_state=config.get(SwitchSchema.CONF_STATE_ADDRESS),
+                respond_to_read=config[CONF_RESPOND_TO_READ],
+                invert=config[SwitchSchema.CONF_INVERT],
+            )
+        )
+        self._attr_entity_category = config.get(CONF_ENTITY_CATEGORY)
+        self._attr_device_class = config.get(CONF_DEVICE_CLASS)
+        self._attr_unique_id = str(self._device.switch.group_address)
+
+
+class KnxUiSwitch(_KnxSwitch):
+    """Representation of a KNX switch configured from UI."""
+
+    def __init__(
+        self, knx_module: KNXModule, unique_id: str, config: dict[str, Any]
+    ) -> None:
+        """Initialize of KNX switch."""
+        super().__init__(
+            device=XknxSwitch(
+                knx_module.xknx,
+                name=config["entity"][CONF_NAME],
+                group_address=config["knx"]["ga_switch"]["write"],
+                group_address_state=[
+                    config["knx"]["ga_switch"]["state"],
+                    *config["knx"]["ga_switch"]["passive"],
+                ],
+                respond_to_read=config["knx"][CONF_RESPOND_TO_READ],
+                invert=config["knx"]["invert"],
+            )
+        )
+        self._attr_entity_category = config["entity"][CONF_ENTITY_CATEGORY]
+        self._attr_unique_id = unique_id
+        if device_info := config["entity"].get("device_info"):
+            self._attr_device_info = DeviceInfo(identifiers={(DOMAIN, device_info)})
+            self._attr_has_entity_name = True
```

Comment:
Why only set `_attr_has_entity_name = True` when there is device info present?

Reply:
You are right, setting it to True by default should behave exactly the same.